Sources/RichTextKit/Attributes/RichTextLayoutManager.swift

```diff
@@ -0,0 +1,46 @@
+import UIKit
+
+public final class RichTextLayoutManager: NSLayoutManager {
+    public override init() {
+        super.init()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public override func processEditing(
+        for textStorage: NSTextStorage,
+        edited editMask: NSTextStorage.EditActions,
+        range newCharRange: NSRange,
+        changeInLength delta: Int,
+        invalidatedRange invalidatedCharRange: NSRange
+    ) {
+        super.processEditing(
+            for: textStorage,
+            edited: editMask,
+            range: newCharRange,
+            changeInLength: delta,
+            invalidatedRange: invalidatedCharRange
+        )
+        drawCustomAttributes(forGlyphRange: textStorage.richTextRange, at: .zero)
+    }
+    
+    private func drawCustomAttributes(forGlyphRange glyphsToShow: NSRange, at origin: CGPoint) {
+        let characterRange = characterRange(forGlyphRange: glyphsToShow, actualGlyphRange: nil)
+        renderLinks(in: characterRange, at: origin)
+    }
+ 
+    private func renderLinks(in characterRange: NSRange, at origin: CGPoint) {
+        textStorage?.enumerateAttribute(
+            .customLink,
+            in: characterRange,
+            using: { [weak textStorage] value, range, _ in
+                if let value = value as? CustomLinkAttributes, let link = value.link {
+                    textStorage?.addAttribute(.link, value: link, range: range)
+                    textStorage?.addAttribute(.foregroundColor, value: value.color, range: range)
```

Comment:
There is downside to this approach. The `processEditing` method is ran each time UITextView is updated, so it constantly sets the foreground color to the custom link attribute... Question is, do we care?

Reply:
Actually this is a bummer. Working on working solution 😅 : 
https://christiantietze.de/posts/2017/11/syntax-highlight-nstextstorage-insertion-point-change/  

cc @danielsaidi if you want to take a look at custom attributes highlighting :)

Reply:
Deleted RichTextLayoutManager.